### ISSUE TITLE
[Feature Request] Support for hash-equal? memoizing macros.

### ISSUE THREAD

**Issue by jgreco** [BourgondAries/memo#1]
This doesn't memoize because hasheq compares object identities:
```
(define/memoize (string-repeat x) (string-append x x))
(string-repeat (string-append "foo" "bar"))
```

Comparing object identities seems like an appropriate default, since eq? is faster than equal?, but it would be nice if there were alternative forms for each of these macros to use equal? hashes.   Maybe something like this:

```
(define/memoize* (string-repeat x) (string-append x x))
(string-repeat (string-append "foo" "bar"))
```

(*-names for the equal? variants is what the [memoize](http://docs.racket-lang.org/memoize/index.html#%28form._%28%28lib._memoize%2Fmain..rkt%29._define%2Fmemo%2A%29%29) package does, but aside this one issue I much prefer your package to that one.  )

**Comment by kstrafe** [comment]
What do you think of `(define/memoize (string-repeat x) #:hash hash ...)` for instance, defaulting to `hasheq` when no such keyword is given.

**Comment by jgreco** [comment]
Perfectly acceptable.

**Comment by kstrafe** [comment]
Implemented in 3ecfa4ad20c38ce97fedaed848d08348e92c56d3, and should be up within 24 hours on the package index.
I'll leave this open for a little while if anything's wrong.
